### PR TITLE
Improve readability of Vagrantfile and cleanup this official image

### PR DESCRIPTION
**Pull request by errordeveloper**

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -1,7 +1,66 @@
...
+SCRIPT
+
+$cleanup = <<SCRIPT
+export DEBIAN_FRONTEND=noninteractive
+## Who the hell thinks official images have to have both of these?
+/etc/init.d/chef-client stop
+/etc/init.d/puppet stop
+apt-get -qq remove puppet chef
+apt-get -qq autoremove
+killall -9 chef-client
```
Comment (rade) [inline Vagrantfile]: !!!! unbelievable !!!!

**Comment by rade** [comment]
Are our test VMs similarly afflicted by that chef+puppet madness?

**Comment by errordeveloper** [comment]
> Are our test VMs similarly afflicted by that chef+puppet madness?

Yes, as they use the same image as this one. The change came with 8326768b9f4914809cef442aaa7a336a1bcfbe44, and I am pretty sure 14.04 image we used before that wasn't affected.

**Comment by rade** [comment]
in that case please fix the test VMs too.

**Comment by paulbellamy** [comment]
@errordeveloper Please cherry-pick https://github.com/paulbellamy/weave/commit/ed9e10540f92496927f4f212dcfd933e0c2f1275

**Comment by errordeveloper** [comment]
@paulbellamy picked.

**Comment by paulbellamy** [comment]
LGTM, then.

**Pull request merged**